astore/server/auth/factory.go

```diff
@@ -83,13 +87,30 @@ func WithCA(fileContent []byte) Modifier {
 			server.log.Warnf("CA file not specified - will operate without certificates")
 			return nil
 		}
-		signer, err := ssh.ParsePrivateKey(fileContent)
+		caPrivateKey, err := ssh.ParseRawPrivateKey(fileContent)
 		if err != nil {
 			return fmt.Errorf("Could not parse CA key - %w", err)
 		}
-		server.caSigner = signer
-		server.marshalledCAPublicKey = ssh.MarshalAuthorizedKey(signer.PublicKey())
-		return nil
+		// TODO(adam): make parsing existing keys cleaner
+		if key, ok := caPrivateKey.(*ed25519.PrivateKey); ok {
```

Comment:
this smells like code that should go in `kcerts`, simple typeswitch that returns kcerts.From... depending on key type.

Reply:
with common MarshalAuthorizedkey